lighthouse-core/test/config/budget-test.js

```diff
@@ -131,6 +137,67 @@ describe('Budget', () => {
       assert.throws(_ => Budget.initializeBudget(budgets),
         /^Error: Invalid timings entry in budget at index 1$/);
     });
+
+    it('throws when budget contains an invalid options entry', () => {
+      budgets[0].options = 'Turtles';
+      assert.throws(_ => Budget.initializeBudget(budgets),
+        /^Error: Invalid options property in budget at index 0$/);
+    });
+  });
+
+  describe('firstPartyHostname validation', () => {
+    it('with valid inputs', () => {
+      const validHostnames = [
+        'yolo.com',
+        'so.many.subdomains.org',
+        '127.0.0.1',
+        'localhost',
+        '*.example.gov.uk',
+        '*.example.com',
+      ];
+      budgets[0].options = {firstPartyHostnames: validHostnames};
+
+      const result = Budget.initializeBudget(budgets);
+      expect(result[0].options.firstPartyHostnames).toEqual(validHostnames);
+    });
+
+    it('validates that input does not include a protocol', () => {
+      budgets[0].options = {firstPartyHostnames: ['https://yolo.com']};
+      assert.throws(_ => Budget.initializeBudget(budgets),
+        /https:\/\/yolo.com is not a valid hostname./);
+    });
+
+    it('validates that input does not include ports', () => {
+      budgets[0].options = {firstPartyHostnames: ['https://yolo.com:8080']};
```

Comment:
```suggestion
      budgets[0].options = {firstPartyHostnames: ['yolo.com:8080']};
```

just to separate protocol from port testing